src/Concordium/Client/Runner.hs

```diff
@@ -1954,15 +1969,16 @@ processContractCmd action baseCfgDir verbose backend =
         Just (InvokerAccount nameOrAddr) -> Just . Types.AddressAccount . naAddr <$> getAccountAddressArg (bcAccountNameMap baseCfg) nameOrAddr
         Just InvokerContract{..} -> Just . Types.AddressContract . ncaAddr <$> getNamedContractAddress (bcContractNameMap baseCfg) icIndexOrName icSubindex
 
-      contrInfo <- withClient backend . withBestBlockHash block $ \bb -> getContractInfo namedContrAddr bb
+      contrInfo <- withClient backend . withBestBlockHash block $ getContractInfo namedContrAddr
       let namedModRef = NamedModuleRef {nmrRef = CI.ciSourceModule contrInfo, nmrNames = []} -- Skip finding nmrNames, as they won't be shown.
 
       let contractName = CI.getContractName contrInfo
       let wasmReceiveName = Wasm.ReceiveName [i|#{contractName}.#{receiveName}|]
-      unless (CI.hasReceiveMethod receiveName contrInfo) $ logFatal [[i|The contract does not have a receive function named '#{receiveName}'.|]]
+
+      receiveName' <- checkAndGetContractReceiveName contrInfo receiveName
```

Comment:
I'd recommend renaming `receiveName'` to something else, or just not having the return value since it will always be the same as `receiveName`.

Reply:
Then please document this, and still the renaming makes sense, since `'` conveys no meaning.

What is happening here is quite subtle, since the fact that the fallback receive name is returned means that that is the name that will be used when looking up parameters/schemas. So this is quite important to get right.

Reply:
As far as I can see it is the convention used when a new value is assigned to a variable. Here you can see another example of it with `invoker'`. The documentation of the `checkAndGetContractReceiveName` already mentions this possibility. Do we still need to document it again here?

Reply:
The only reason to use `'` is that when I try to shadow the variable name `receiveName` the compiler gives a warning. Shadowing is something quite common in Rust, even in our sources. Apparently, one of us misses a point here. Anyway, I'll rename it.

Reply:
The reason shadowing is really bad in Haskell is that a lot of the bindings are implicitly recursive. For example
```
f x = 
   let x = someFunction x
   in someOtherFunction x
```
does not have the meaning it would have in Rust, or basically any other language.  Similarly with a `where` clause.

So that is why shadowing is a warning that is enabled by default. It is very error prone to shadow variable names in Haskell, especially when one works in a mix of settings.

One "easy" solution to this is to add `'` to variable names. But the problem with that is that it is easy to write, but the `'` has no meaning to the reader and so is hard to read, and the `'` is also easy to miss. When the code is very generic that can be fine, since a more meaningful name maybe does not exist, but in this case the code is very specific, so the variable can be named in a way that makes it easier to understand the code later on. And we should strive to have the code easy to read, and as hard to misinterpret as possible.

Some of our codebase unfortunately has `'` in the names where a better name could exist (and I am guilty of using this many times), but that is not a reason to perpetuate that bad practice.